### PR TITLE
Kyber: add F* proofs

### PR DESCRIPTION
**Pull request by W95Psp**
This PR adds a new directory `extraction-edited` to `proofs/fstar/` with a set of proof for the Kyber implementation.
The code in this new directory started as a snapshot of the generated F* code in `proofs/fstar/extraction` but has been hand edited with annotations, invariants, pre- and post-conditions, as well as a functional specification of Kyber.

The proofs currently check the following:
- all the modules are proved to be panic-free
- modular arithmetic are functionally verified (`Libcrux.Kem.Kyber.Arithmetic`) 
- constant-time operations are functionally verified (`Libcrux.Kem.Kyber.Constant_time_ops`) 
- the functions implementing IND-CPA encryption, and the IND-CCA KEM are functionally verified
- part of serialization and NTT are functionally verified; in particular, we prove that all range assumptions hold, and that there are no arithmetic overflows
- secret independence is partially verified (in particular, division with secret inputs is forbidden)

However, some parts remain, and we are adding open issues to track them and will address them in future PRs.
- full functional correctness for NTT, matrix operations, and serialization
- secret independence for the whole code

In further future work, we will seek to address the following:
- minimize diffs between `extraction-edited` and `extraction` by propagating pre- and post- conditions and meaningful annotations to the source Rust code
- put the proofs and the proof-related patch in `extraction-edited` under CI
- re-verify any recent changes to the Rust code

### CODE REVIEW

File: specs/kyber/src/ind_cpa.rs [hunk 1]
```diff
@@ -188,9 +188,6 @@ fn encode_and_compress_u(input: KyberVector) -> Vec<u8> {
     out
 }
 
-/// This function implements <strong>Algorithm 13</strong> of the
```
Comment (franziskuskiefer) [inline specs/kyber/src/ind_cpa.rs:191]: Is there a reason to remove this?

File: src/kem/kyber/helper.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#[cfg(not(hax))]
+#[macro_export]
+macro_rules! cloop {
```
Comment (franziskuskiefer) [inline src/kem/kyber/helper.rs:3]: This shouldn't be a change. Looks like the branch is outdated.

**Review by franziskuskiefer — COMMENTED** [review]
Looks like this needs to be rebased and cleaned up a little.
Is there another reason this is a draft?

**Comment by W95Psp** [comment]
I disabled CI for now on `extraction-edited`, see #168.

**Review by franziskuskiefer — APPROVED** [review]
Ok, let's get this in.
Not sure if you want to update the `PROOFS.md` before merging.

**Comment by karthikbhargavan** [comment]
> Ok, let's get this in. Not sure if you want to update the `PROOFS.md` before merging.

That file was out of date, so I removed it. The comments on this PR and the remaining issues are the ones that are most relevant.

**Pull request merged**